File: server/src/main/java/me/ajfleming/tworoomsio/storage/GameCache.java
package me.ajfleming.tworoomsio.storage;

import java.util.Optional;
import me.ajfleming.tworoomsio.exception.GameException;
import me.ajfleming.tworoomsio.model.Game;

public interface GameCache {

  /**
   * Returns the Game object mapped to the GameId
   * @param gameId The Unique Identifier of the Game
   * @return The Game Object representing the game
   */
  Game getGame(String gameId) throws GameException;

  /**
   * Updates the GameCache with an update to date version of the game
   * @param game The updated game object
   */
  void addGame(Game game);
}

File: kotlin_server/src/main/kotlin/me/ajfleming/tworoomsio/storage/GameCache.kt
package me.ajfleming.tworoomsio.storage

import me.ajfleming.tworoomsio.model.Game
import me.ajfleming.tworoomsio.model.Player

interface GameCache {

    fun getGame(gameId: String) : Game?

    fun getGameByJoinCode(joinCode: String) : Game?

    fun addGame(game: Game)

    fun getGamesPlayerIsIn(playerToken: Player) : List<Game>

    fun deleteGame(gameId: String)
}